superset-frontend/src/SqlLab/components/TableElement.jsx

```diff
@@ -52,59 +51,40 @@ const StyledSpan = styled.span`
   cursor: pointer;
 `;
 
-class TableElement extends React.PureComponent {
-  constructor(props) {
-    super(props);
-    this.state = {
-      sortColumns: false,
-      expanded: true,
-      hovered: false,
-    };
-    this.removeFromStore = this.removeFromStore.bind(this);
-    this.toggleSortColumns = this.toggleSortColumns.bind(this);
-    this.removeTable = this.removeTable.bind(this);
-    this.setHover = this.setHover.bind(this);
-  }
-
-  setHover(hovered) {
-    this.setState({ hovered });
-  }
+function TableElement({ actions, table, timeout }) {
+  const [state, setState] = useState({
```

Comment:
Break these into individual hooks

Reply:
Completed at [`this commit`](https://github.com/apache/superset/pull/12949/commits/28bb86ab8231a8c03490d6e69c269287fde472d2)